plugins/repository-azure/src/main/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceImpl.java

```diff
@@ -35,54 +37,92 @@
 import java.io.OutputStream;
 import java.net.URI;
 import java.net.URISyntaxException;
+import java.util.Hashtable;
 import java.util.Map;
 
-import static org.elasticsearch.cloud.azure.storage.AzureStorageService.Storage.*;
-
-/**
- *
- */
 public class AzureStorageServiceImpl extends AbstractLifecycleComponent<AzureStorageServiceImpl>
     implements AzureStorageService {
 
-    private final String account;
-    private final String key;
-    private final String blob;
-
-    private CloudBlobClient client;
+    final AzureStorageSettings primaryStorageSettings;
+    final Map<String, AzureStorageSettings> secondariesStorageSettings;
 
+    final Map<String, CloudBlobClient> clients;
+    
     @Inject
     public AzureStorageServiceImpl(Settings settings) {
         super(settings);
-        // We try to load storage API settings from `cloud.azure.`
-        account = settings.get(ACCOUNT);
-        key = settings.get(KEY);
-        blob = "https://" + account + ".blob.core.windows.net/";
 
+        Tuple<AzureStorageSettings, Map<String, AzureStorageSettings>> storageSettings = AzureStorageSettings.parse(settings);
+        this.primaryStorageSettings = storageSettings.v1();
+        this.secondariesStorageSettings = storageSettings.v2();
+
+        this.clients = new Hashtable<>();
+    }
+
+    void createClient(AzureStorageSettings azureStorageSettings) {
         try {
-            if (account != null) {
-                logger.trace("creating new Azure storage client using account [{}], key [{}], blob [{}]", account, key, blob);
+            String blob = "https://" + azureStorageSettings.getAccount() + ".blob.core.windows.net/";
```

Comment:
Since the account settings are supplied by user, I would feels better if we used URI to build this string. This way we will have at least some basic validation of the things that go into this URL.